test/arrays.jl

```diff
@@ -139,58 +140,174 @@ The following two testsets test jacobians for symbolic functions of symbolic arr
 
     fun_genf = build_function(ex, x, expression=Val{false})
-    @test_broken fun_genf(x0) == A*x0# UndefVarError: foo not defined
+    @test_broken fun_genf(x0) == A * x0# UndefVarError: foo not defined
 
     # Generate an expression instead and eval it manually
     fun_ex = build_function(ex, x, expression=Val{true})
     fun_eval = eval(fun_ex)
-    @test fun_eval(x0) == foo(x0) 
+    @test fun_eval(x0) == foo(x0)
 
     # Try to provide the hidden argument `expression_module` to solve the scoping issue
     @test_skip begin
         fun_genf = build_function(ex, x, expression=Val{false}, expression_module=Main) # UndefVarError: #_RGF_ModTag not defined
-        fun_genf(x0) == A*x0 
+        fun_genf(x0) == A * x0
     end
 
     ## Jacobians
     @test Symbolics.value.(Symbolics.jacobian(foo(x), x)) == A
-    @test_throws ErrorException Symbolics.value.(Symbolics.jacobian(ex , x))
+    @test_throws ErrorException Symbolics.value.(Symbolics.jacobian(ex, x))
 end
 
 
 @testset "Functions and Jacobians using manual @wrapped" begin
     @variables x[1:n]
 
     x0 = randn(n)
-    @test foo(x0) == A*x0
+    @test foo(x0) == A * x0
     ex = foo(x)
 
     @test shape(ex) == shape(x)
 
     fun_genf = build_function(ex, x, expression=Val{false})
-    @test fun_genf(x0) == A*x0
+    @test fun_genf(x0) == A * x0
 
     # Generate an expression instead and eval it manually
     fun_ex = build_function(ex, x, expression=Val{true})
     fun_eval = eval(fun_ex)
-    @test fun_eval(x0) == foo(x0) 
+    @test fun_eval(x0) == foo(x0)
 
     ## Jacobians
     @test value.(jacobian(foo(x), x)) == A
-    @test value.(jacobian(ex , x)) == A
+    @test value.(jacobian(ex, x)) == A
 end
 
 @testset "Rules" begin
-    @variables X[1:10,1:5] Y[1:5, 1:10] b[1:10];
-    r = @rule ((~A*~B)*~C) => (~A*(~B*~C)) where size(~A,1)*size(~B,2) > size(~B,1)*size(~C,2)
-    @test isequal(r(unwrap((X*Y)*b)), unwrap(X*(Y*b)))
+    @variables X[1:10, 1:5] Y[1:5, 1:10] b[1:10]
+    r = @rule ((~A * ~B) * ~C) => (~A * (~B * ~C)) where size(~A, 1) * size(~B, 2) >size(~B, 1)  * size(~C, 2)
+    @test isequal(r(unwrap((X * Y) * b)), unwrap(X * (Y * b)))
+end
+
+@testset "2D Diffusion Composed With Stencil Interface" begin
+    n = rand(8:32)
+
+    @variables u[1:n, 1:n]
+    @makearray v[1:n, 1:n] begin
+        #interior
+        v[2:end-1, 2:end-1] => @arrayop (i, j) u[i-1, j] + u[i+1, j] + u[i, j-1] + u[i, j+1] - 4 * u[i, j]
+        #BCs
+        v[1, 1:end] => 0.0
+        v[n, 1:end] => 0.0
+        v[1:end, 1] => 0.0
+        v[1:end, n] => 0.0
+    end
+
+    #2D Diffusion composed
+    @makearray ucx[1:n, 1:n] begin
+        ucx[1:end, 1:end] => 0.0 # fill zeros
+        ucx[2:end-1, 2:end-1] => @arrayop (i, j) u[i-1, j] + u[i+1, j] - 2 * u[i, j] (j in 2:n-1)
+    end
+
+    @makearray ucy[1:n, 1:n] begin
+        ucy[1:end, 1:end] => 0.0 # fill zeros
+        ucy[2:end-1, 2:end-1] => @arrayop (i, j) u[i, j-1] + u[i, j+1] - 2 * u[i, j] (i in 2:n-1)
```

Comment:
@xtalax here you need to specify that `i` is between 2 to n-1, because Symbolics will infer the size of the array op to be `1:n` if you don't and this assignment will not work. I added a nice descriptive error message for this.